x-pack/plugins/security/server/lib/authorization/register_privileges_with_cluster.test.js

```diff
@@ -138,21 +147,37 @@ const registerPrivilegesWithClusterTest = (description, {
   test(description, async () => {
     const mockServer = createMockServer();
     const mockCallWithInternalUser = registerMockCallWithInternalUser()
-      .mockImplementationOnce(async () => {
-        if (throwErrorWhenGettingPrivileges) {
-          throw throwErrorWhenGettingPrivileges;
-        }
+      .mockImplementation((api) => {
+        switch(api) {
+          case 'shield.getPrivilege': {
+            if (throwErrorWhenGettingPrivileges) {
+              throw throwErrorWhenGettingPrivileges;
+            }
 
-        // ES returns an empty object if we don't have any privileges
-        if (!existingPrivileges) {
-          return {};
-        }
+            // ES returns an empty object if we don't have any privileges
+            if (!existingPrivileges) {
+              return {};
+            }
+
+            return existingPrivileges;
+          }
+          case 'shield.deletePrivilege': {
+            if (throwErrorWhenDeletingPrivileges) {
+              throw throwErrorWhenDeletingPrivileges;
+            }
 
-        return existingPrivileges;
-      })
-      .mockImplementationOnce(async () => {
-        if (throwErrorWhenPuttingPrivileges) {
-          throw throwErrorWhenPuttingPrivileges;
+            break;
+          }
+          case 'shield.postPrivileges': {
+            if (throwErrorWhenPuttingPrivileges) {
+              throw throwErrorWhenPuttingPrivileges;
+            }
+
+            return;
+          }
+          default: {
+            expect(true).toBe(false);
```

Comment:
nit: test failures here might be hard to debug. Consider throwing an explicit exception about this unhandled scenario.

Reply:
If we just threw an explicit exception here, it'd potentially be caught by the code using the callWithRequest and lead to rather difficult to understand behavior. I wanted to fail the test explicitly, and this was the most clear solution that I could come up with, even though it isn't really clear at all. I thought about also doing the equivalent of `expect(api).toBeInArray(['shield.getPrivilege', 'shield.postPrivileges', 'shield.deletePrivilege']` but figured that'd be equally confusing, and potentially more-so, so I gave up and did this.